Разбил приложение на модули

rvkhlystov opened the pull request.

rvkhlystov committed:
Разбил приложение на модули

RodinDenis commented on server/pom.xml:
```diff
@@ -0,0 +1,29 @@
+            <version>1.0</version>
+            <scope>compile</scope>
+        </dependency>
+        <dependency>
```
лучше общий код оставить в common и убрать зависимость от app

rvkhlystov replied:
Спасибо! Значит не зря я весь вечер потратил на рефакторинг, чтобы максимально уйти от зависимостей между app и server. Подумаю еще, избавиться от оставшихся зависимостей

rvkhlystov replied:
Все оказалось проще. связь только из-за теста. превращу тест в модульный и связь уйдет

rvkhlystov replied:
замечание устранено

rvkhlystov committed:
Добавил в app и server классы Main (пока без наполнения)

rvkhlystov committed:
Заменил в ряде классов переопределение Equals на аннотацию @EqualsAndHashCode

rvkhlystov committed:
Отвязал тест сервера от app

rvkhlystov committed:
Сделал:
1. Класс Interaction перенес из common в приложение
2. В модулях приложение и сервер выделил сервисы и контролеры
3. Реализовал контролеры

Проблема:
При получении сервером от приложения rest запроса на проведение платежа он не может его десериализовать

rvkhlystov committed:
Сделал:
1. Реализовал обходным путем передачу платежа: для передачи платежа приложение его разбирает и складывает значения полей в HashMap,после чего отправляет HashMap RESTом на сервер; сервер после приема запроса из HashMap собирает объект класса Payment и дальше обрабатывает его. Для реализации был создал новый общий класс CreatorTransferPayment с двумя методами.

Необходимо обходное решение заменить на постоянное, разобравшись почему не проходит десериализация запроса с платежом (судя по всему из-за наличия в объекте Payment объекта PhoneNumberRussian)

rvkhlystov committed:
Сделал:
1. Добавил логирование в контроллеры
2. Закоментил в pom-файлах ненужные зависимости

Необходимо обходное решение заменить на постоянное, разобравшись почему не проходит десериализация запроса с платежом (судя по всему из-за наличия в объекте Payment объекта PhoneNumberRussian)
